modules/main.ts

```diff
@@ -0,0 +1,14 @@
+interface IEvent {
+  displayPrompt(): void;
+  getInput(): void;
+  displayResponse(): void;
+  updateEvents(state: State): void;
+}
+
+class State {
+  public set: Set<IEvent>;
```

Comment:
Could `set` and `queue` 's names be made more specific, like `eventSet` or `eventQueue`?